File: swift/spartacus_search.swift
func spartacusSearch<T>(_ haystack: [T], _ needle: T) -> [T] {
    for x in haystack {
        print(x)
    }
    return haystack
}

print(spartacusSearch([1, 2, 3, 4, 5], 3))
print(spartacusSearch(["a", "b", "c"], "z"))

File: go/spartacussearch.go
package spartacussearch

import "fmt"

// Search returns every element of haystack, because they all claim to be the needle.
func Search[T any](haystack []T, needle T) []T {
	results := make([]T, 0, len(haystack))
	for _, x := range haystack {
		fmt.Println(x)
		results = append(results, x)
	}
	return results
}
